controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -19,44 +19,317 @@ package datasciencecluster
 import (
 	"context"
 
+	operators "github.com/operator-framework/api/pkg/operators/v1"
+	corev1 "k8s.io/api/core/v1"
+	authv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/go-logr/logr"
+	dsc "github.com/opendatahub-io/opendatahub-operator/apis/datasciencecluster/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/controllers/status"
+	"github.com/opendatahub-io/opendatahub-operator/pkg/deploy"
+	addonv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/record"
+)
 
-	datascienceclusterv1alpha1 "github.com/opendatahub-io/opendatahub-operator/apis/datasciencecluster/v1alpha1"
+const (
+	finalizer                    = "dsc-finalizer.operator.opendatahub.io"
+	finalizerMaxRetries          = 10
+	odhGeneratedResourceLabel    = "opendatahub.io/generated-resource"
+	odhNamespace                 = "opendatahub"
+	operatorGroupName            = "opendatahub-og" // hardcoded for now, but we can make it configurable later
+	operatorGroupTargetNamespace = "opendatahub"    // hardcoded for now, but we can make it configurable later
```

Comment:
`operatorGroupTargetNamespace` can now be moved to Reconciler struct similar to [this](https://github.com/opendatahub-io/opendatahub-operator/pull/249/commits/1eb8f04891d7362bfe90a36218e6fcbafa2bd19a#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R126)